src/wp-includes/link-template.php

```diff
@@ -3495,9 +3495,15 @@ function get_site_url( $blog_id = null, $path = '', $scheme = null ) {
 	if ( empty( $blog_id ) || ! is_multisite() ) {
 		$url = get_option( 'siteurl' );
 	} else {
-		switch_to_blog( $blog_id );
-		$url = get_option( 'siteurl' );
-		restore_current_blog();
+		// Use WP_Site_Query to get the site URL.
+		$site = get_site( $blog_id );
+
+		$url = $site->domain . $site->path;
+
+		// If the network has a custom domain, use that instead.
+		if ( \get_network()->domain !== $site->domain ) {
```

Comment:
Why `\` use here? Code did not use it in any code references

Reply:
Sorry, my bad. It's in my fingers, add them automatically because I always use namespace. I'll remove it i a few sec.